### ISSUE TITLE
[CVE] Bump `tsd` to `0.21.0` to remove dependency on `got`

### ISSUE THREAD

**Issue by joshuarrrr** [opensearch-project/opensearch-js#258]
From https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1764, CVE-2022-33987 can be remediated by removing or updating `got`. For this project, the easiest fix is to update `tsd`, which no longer has a `got` dependency.

```
yarn why got
yarn why v1.22.18
[1/4] Why do we have the module "got"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "got@9.6.0"
info Reasons this module exists
   - "tsd#update-notifier#latest-version#package-json" depends on it
   - Hoisted from "tsd#update-notifier#latest-version#package-json#got"
Done in 0.31s.
```

`tsd v0.17.0` removes dependency chain:
- `update-notifier`
- `latest-version`
- `package-json`
- `got`

**Comment by joshuarrrr** [comment]
Note that this issue is blocked until we update the node dependency of this repo - `0.7.0` requires node 12: https://github.com/SamVerschueren/tsd/releases/tag/v0.17.0

**Comment by VachaShah** [comment]
Closed via #272
